llama-index-integrations/selectors/llama-index-selectors-notdiamond/llama_index/selectors/notdiamond/base.py

```diff
@@ -0,0 +1,168 @@
+import logging
+import os
+from typing import Sequence
+
+from llama_index.core.llms.llm import LLM
+from llama_index.core.schema import QueryBundle
+from llama_index.core.tools.types import ToolMetadata
+from llama_index.core.base.base_selector import SelectorResult
+from llama_index.core.selectors import LLMSingleSelector
+from llama_index.core.selectors.llm_selectors import _build_choices_text
+
+from notdiamond import NotDiamond, LLMConfig, Metric
+
+LOGGER = logging.getLogger(__name__)
+LOGGER.setLevel(logging.WARNING)
+
+
+class NotDiamondSelectorResult(SelectorResult):
+    """A single selection of a choice provided by Not Diamond."""
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    session_id: str
+    llm: LLMConfig
+
+    @classmethod
+    def from_selector_result(
+        cls, selector_result: SelectorResult, session_id: str, best_llm: LLMConfig
+    ) -> "NotDiamondSelectorResult":
+        return cls(session_id=session_id, llm=best_llm, **selector_result.dict())
+
+
+class NotDiamondSelector(LLMSingleSelector):
+    def __init__(
+        self,
+        client: NotDiamond,
+        metric: Metric = None,
+        timeout: int = 10,
+        api_key: str = None,
+        *args,
+        **kwargs,
+    ):
+        # Not needed - we will route using our own client based on the query prompt
+        # Add @property for _llm here
+        _encap_selector = LLMSingleSelector.from_defaults()
```

Comment:
Note we don't use this encapsulated selector beyond initialization. There's probably a cleaner way to handle this.